src/spi.rs

```diff
@@ -414,14 +438,60 @@ impl<'d> SpiMasterDriver<'d> {
             quadwp_io_num: -1,
             quadhd_io_num: -1,
 
-            max_transfer_sz: config.dma.max_transfer_size() as i32,
+            max_transfer_sz: max_transfer_sz as i32,
             ..Default::default()
         };
 
-        esp!(unsafe { spi_bus_initialize(SPI::device(), &bus_config, config.dma.into()) })?;
+        esp!(unsafe { spi_bus_initialize(SPI::device(), &bus_config, dma_chan) })?;
+        Ok(max_transfer_sz)
+    }
+}
+
+impl<'d> Drop for SpiMasterDriver<'d> {
+    fn drop(&mut self) {
+        // should we drop without checking the lock?
+        let guard = self.lock.enter();
+        esp!(unsafe { spi_bus_free(self.handle) }).unwrap();
+    }
+}
+
+pub struct EspSpiDevice<'d, T> {
+    handle: RefCell<Option<spi_device_handle_t>>,
+    cs_pin_ref: PeripheralRef<'d, AnyOutputPin>,
+    config: config::Config,
+    con: spi_device_interface_config_t,
+    driver: T,
+    _p: PhantomData<&'d ()>,
```

Comment:
No need for `PhantomData` since `PeripheralRef` above makes use of `'d`

Reply:
this PhantomData is still :ghost: ghostly for me. At least it doesn't use any space :smile_cat:

Reply:
Yeah it doesn't but anyone who reviews this code will now take a few moments to question its use, which is less than ideal.

Reply:
Or the other way around (as I did in all other drivers): demote the `cs_pin_ref` to a single u8 (actually - do you even need it and if not then why keeping it around in the first place?) `_p` takes care of the lifetime already anyway.